src/meltano/core/plugin/singer/tap.py

```diff
@@ -557,6 +559,7 @@ def apply_catalog_rules(  # noqa: WPS213, WPS231, WPS210
                 SchemaExecutor(schema_rules).visit(catalog)
 
             if metadata_rules:
+                self.warn_property_not_found(metadata_rules, catalog)
```

Comment:
Thanks @haleemur! IIUC, this checks the metadata rules against the stream's schema so I suggested a slightly different message when a property can't be found.